lib/babe/build.go

```diff
@@ -246,3 +248,63 @@ func extrinsicsToBody(txs []*transaction.ValidTransaction) (*types.Body, error)
 
 	return types.NewBodyFromExtrinsics(extrinsics)
 }
+
+func determineError(res []byte) string {
+	log.Error("[babe] build block apply extrinsic", "error", res)
+	var errTxt strings.Builder
+
+	// when res[0] == 0x01 it is an apply error
+	if res[0] == 1 {
+		errTxt.WriteString("Apply error, type: ")
+		if bytes.Equal(res[1:], []byte{0}) {
+			errTxt.WriteString("NoPermission")
+		}
+		if bytes.Equal(res[1:], []byte{1}) {
+			errTxt.WriteString("BadState")
+		}
+		if bytes.Equal(res[1:], []byte{2}) {
+			errTxt.WriteString("Validity")
+		}
+		if bytes.Equal(res[1:], []byte{2, 0, 0}) {
+			errTxt.WriteString("Call")
+		}
+		if bytes.Equal(res[1:], []byte{2, 0, 1}) {
+			errTxt.WriteString("Payment")
+		}
+		if bytes.Equal(res[1:], []byte{2, 0, 2}) {
+			errTxt.WriteString("Future")
+		}
+		if bytes.Equal(res[1:], []byte{2, 0, 3}) {
+			errTxt.WriteString("Stale")
+		}
+		if bytes.Equal(res[1:], []byte{2, 0, 4}) {
+			errTxt.WriteString("BadProof")
+		}
+		if bytes.Equal(res[1:], []byte{2, 0, 5}) {
+			errTxt.WriteString("AncientBirthBlock")
+		}
+		if bytes.Equal(res[1:], []byte{2, 0, 6}) {
+			errTxt.WriteString("ExhaustsResources")
+		}
```

Comment:
Is this check enough to close #504?  I've determined that the block is full, but do I need to do more, or is it enough just to include this information in error text?

Reply:
I think this is good enough